airflow/dag_processing/processor.py

```diff
@@ -149,8 +149,8 @@ def _run_file_processor(
 
         try:
             # redirect stdout/stderr to log
-            with redirect_stdout(StreamLogWriter(log, logging.INFO)), redirect_stderr(
-                StreamLogWriter(log, logging.WARN)
+            with redirect_stdout(StreamLogWriter(log, logging.INFO)), (
+                redirect_stderr(StreamLogWriter(log, logging.WARN))
             ), Stats.timer() as timer:
```

Comment:
This does not work because [you can’t use parentheses in a `with` statement like anywhere else in the language, until 3.10](https://bugs.python.org/issue12782) 😞 

A popular workaround is

```python
import contextlib

with contextlib.ExitStack() as stack:
    stack.enter_context(redirect_stdout(StreamLogWriter(log, logging.INFO)))
    stack.enter_context(redirect_stderr(StreamLogWriter(log, logging.WARN)))
    timer = stack.enter_context(Stats.timer())
```

Reply:
That is...unfortunate. This change was made to make flake8 happy for line-length validation. Now that the Mypy ignore comments are gone, we could revert this change. WDYT?

Reply:
> Although I’m not sure why we need `encoding` exactly; maybe we could remove this test entirely?

I looked at the history and the reason is that some of the internals of subprocess use  stderr.encoding and stdout.enconding and we are effectvely replace stderr and stdout with StreamLogWriter (which has other bad consequences like infinte redirect when someone wants to also log stufff to stderr/stdout - buit this is a different topic - #20500 )

This is an excerpt from subprcess.py where it woudl fail:
```

# Translate newlines, if requested.
            # This also turns bytes into strings.
            if self.text_mode:
                if stdout is not None:
                    stdout = self._translate_newlines(stdout,
                                                      self.stdout.encoding,
                                                      self.stdout.errors)
                if stderr is not None:
                    stderr = self._translate_newlines(stderr,
                                                      self.stderr.encoding,
                                                      self.stderr.errors)

            return (stdout, stderr)
```

On the other hand -  we should not really IMHO set it to None (though it results in "utf-8" being used in `decode` of bytes)  but likely with the `sys.getdefaultencoding()`

Reply:
(and BTW  -yeah, we should rewrite big part of our logging implementation to make it sane).